README.md

```diff
@@ -1,2 +1,131 @@
-# machine-learning-integration-plugin
-A CRS plugin to use machine learning together with the rule set.
+# OWASP ModSecurity Core Rule Set - Machine Learning Plugin
+
+## Description
+
+This is a plugin that serves to integrate machine learning support to the CRS.
+
+The plugin consists of 3 parts -
+1. Plugin rules
+2. Lua script
+3. Flask server
+
+### Plugin Rules
+The plugin contains rules which call the Lua script and pass or block the request based on the status returned to the rule.
+
+### Lua Script
+The lua script receives the request from the rule, reads the data sent in the request and reconstructs the request to form a http POST request to be sent to the flask server ```ml_model_server```. 
+After receiving the response from the server, the lua script sends the status response back to the CRS plugin. 
+
```

Comment:
I'm missing the documentation about format of the data which is script sending to ML server.

Reply:
The http post request to the ML server is being sent in the form of a string. I have added this to the documentation

Reply:
But how that string looks like? Image you are developing the server side software and needs to process data sends by your script.